server/src/test/java/org/elasticsearch/index/mapper/NestedObjectMapperTests.java

```diff
@@ -742,4 +742,40 @@ public void testReorderParent() throws IOException {
         assertThat(doc.docs().get(1).get("nested1.field2"), equalTo("4"));
         assertThat(doc.docs().get(2).get("field"), equalTo("value"));
     }
+
+    public void testMergeNestedMappings() throws  IOException {
+        MapperService mapperService = createIndex("index1", Settings.EMPTY, jsonBuilder().startObject()
```

Comment:
Instead of creating two different indices, then merging the object mappers, I think it would be more realistic to create one index, then call `MapperService.merge` with a mapping update. You can look at `testLimitOfNestedFieldsPerIndex` in this class for an example of how to perform the merge.